src/content-handlers/iiif/modules/uv-mediaelementcenterpanel-module/MediaElementCenterPanel.ts

```diff
@@ -59,12 +60,42 @@ export class MediaElementCenterPanel extends CenterPanel<
     });
 
     this.extensionHost.subscribe(IIIFEvents.SET_TARGET, (target: TFragment) => {
-      let t = target.t;
+      // Clear any existing timeout
+      if (this.pauseTimeoutId !== null) {
+        clearTimeout(this.pauseTimeoutId);
+        this.pauseTimeoutId = null;
+      }
+
+      let t: number | [number, number] = target.t;
       if (Array.isArray(t)) {
-        t = t[0];
+        if ((t as [number] | [number, number]).length === 1) {
+          t = t[0];
+        } else {
+          const [startTime, endTime] = t;
+
+          if (endTime <= startTime) {
+            console.error("endTime must be greater than startTime");
+            return;
+          }
+
+          this.player.setCurrentTime(startTime);
+
+          if (this.config.options.autoPlayOnSetTarget) {
+            this.player.play();
+
+            const duration = (endTime - startTime) * 1000;
+
+            this.pauseTimeoutId = setTimeout(() => {
+              this.player.pause();
+              this.pauseTimeoutId = null; // Clear the timeout ID after execution
+            }, duration);
+          }
+
+          return;
+        }
       }
-      that.player.setCurrentTime(t);
-      that.player.play();
+      this.player.setCurrentTime(t);
+      this.player.play();
```

Comment:
Do you also need to account for `this.config.options.autoPlayOnSetTarget` here, or is there a reason this case is different?

Reply:
have pushed a fix